runtime/doc/lsp.txt

```diff
@@ -1579,7 +1579,9 @@ the current buffer.
 
 
     Fields: ~
-      • {silent}?  (`boolean`)
+      • {silent}?      (`boolean`)
+      • {keep_empty}?  (`boolean`) Show clients in hover that produced no
```

Comment:
special case boolean flags are strongly discouraged.

if the problem solved by this pr is actually a problem, it should just be the default behavior. for other caess, a callback should allow the user to modify results arbitrarily, which avoids even more boolean flags in the future.